server/src/main/java/io/crate/execution/engine/indexing/IndexWriterProjector.java

```diff
@@ -185,7 +186,8 @@ public String value() {
             if (value == null) {
                 return null;
             }
-            Map<String, Object> filteredMap = XContentMapValues.filter(value, includes, excludes);
+            assert value instanceof LinkedHashMap<String, Object> : "the raw source order should be preserved";
+            LinkedHashMap<String, Object> filteredMap = (LinkedHashMap<String, Object>) XContentMapValues.filter(value, includes, excludes);
```

Comment:
Isn't the assertion enough? why do you need to cast as well?

Reply:
I just wanted to give more context but I think you are right that assert should be enough.